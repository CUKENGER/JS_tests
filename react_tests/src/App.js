
import { useEffect, useState } from 'react';
import './App.css';
import { BrowserRouter, Routes, Route, Link } from 'react-router-dom';
import Main from './pages/Main';
import About from './pages/About';

// function App() {

// 	const [data, setData] = useState(null)
// 	const [toggle, setToggle] = useState(false)
// 	const [value, setValue] = useState('')

// 	const onClick = () => setToggle(prev => !prev)
		

// 	useEffect(() => {
// 		setTimeout(() => {
// 			setData({})
// 		}, 100)
// 	}, [])

// 	return (
		
// 		<div className="App">
// 			<h1 data-testid="input-elem">{value}</h1>
// 			{toggle === true && <div data-testid="toggle-element">toggle</div>}
// 			{data && <div>data</div>}
// 			<h1>Hello World</h1>
// 			<button data-testid="toggle-btn" onClick={onClick}>Click me</button>
// 			<input onChange={e => setValue(e.target.value)} type="text" placeholder='Enter text'/>
// 		</div>
// 	);
// }

// export default App;

const App = () => {
	

	return (
		<BrowserRouter>
			<Link to="/" data-testid="main-link">main</Link>
			<Link to="/about" data-testid="about-link">about</Link>
			<Routes>
				<Route path='/' element={<Main/>}/>
				<Route path='/about' element={<About/>}/>
			</Routes>
		</BrowserRouter>
	)
};

export default App;
