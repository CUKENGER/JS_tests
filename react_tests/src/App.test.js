import { render, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from './App';


describe('test app', () => {

	test('renders learn react link', () => {
		render(<App />);
		const hellWorldElement = screen.getByText(/Hello world/i);
		const btn = screen.getByRole('button');
		const input = screen.getByPlaceholderText(/Enter text/i);
		expect(hellWorldElement).toBeInTheDocument();
		expect(btn).toBeInTheDocument();
		expect(input).toMatchSnapshot()
		// expect(input).toBeInTheDocument();
		// screen.debug()
	})

	test('tester', async () => {
		render(<App/>)
		// const hellu = screen.queryByText(/hello2/i)
		// expect(hellu).toBeNull()

		const finder = await screen.findByText(/data/i)
		expect(finder).toBeInTheDocument()
	})

	test('btn', () => {
		render(<App/>)
		const btn = screen.getByTestId("toggle-btn")
		// const toggleDiv = screen.queryByTestId("toggle-element")
		expect(screen.queryByTestId("toggle-element")).toBeNull()
		fireEvent.click(btn)
		expect(screen.queryByTestId("toggle-element")).toBeInTheDocument()
	})


	test('input', async () => {
		render(<App/>)
		const input = screen.getByTestId("input-elem")
		expect(screen.getByTestId("input-elem")).toContainHTML('')


		// fireEvent.change(input, {
			// target: {value: '123123'}
		// })

		await userEvent.type(input, '123123')
		expect(screen.getByTestId("input-elem")).toContainHTML('123123')


	})

});
