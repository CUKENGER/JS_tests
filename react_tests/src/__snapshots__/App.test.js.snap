// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<input
  placeholder="Enter text"
  type="text"
/>
`;

exports[`test app renders learn react link 1`] = `
<input
  placeholder="Enter text"
  type="text"
/>
`;
